framework/include/utils/InputParameterWarehouse.h

```diff
@@ -127,4 +136,61 @@ class InputParameterWarehouse : public Warehouse<InputParameters>
   friend class SolidMaterialProperties;
 };
 
+
+template<typename T>
+ControllableParameter<T>
+InputParameterWarehouse::getControllableParameter(const MooseObjectParameterName & input, bool mark_as_controlled /*=false*/)
+{
+
+  // The ControllableParameter object to return
+  ControllableParameter<T> output;
+
+  // Vector of desired parameters
+  std::vector<MooseObjectParameterName> params(1, input);
+  std::map<MooseObjectParameterName, std::vector<MooseObjectParameterName> >::const_iterator link_it = _input_parameter_links.find(input);
+  if (link_it != _input_parameter_links.end())
+    params.insert(params.end(), link_it->second.begin(), link_it->second.end());
+
+  // Parameter object iterator
+  std::multimap<MooseObjectName, MooseSharedPointer<InputParameters> >::const_iterator iter;
+
+  // Loop over all threads
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); ++tid)
+  {
+    // Loop over all InputParameter objects
+    for (iter = _input_parameters[tid].begin(); iter != _input_parameters[tid].end(); ++iter)
+    {
+      // Loop of all desired params
+      for (std::vector<MooseObjectParameterName>::iterator it = params.begin(); it != params.end(); ++it)
+      {
+        // If the desired object name does not match the current object name, move on
+        MooseObjectParameterName desired = *it;
+
+        if (desired != iter->first)
+          continue;
+
+        // If the parameter is valid and controllable update the output vector with a pointer to the parameter
+        if (iter->second->libMesh::Parameters::have_parameter<T>(desired.parameter()))
+        {
+          // Do not allow non-controllable types to be controlled
+          if (!iter->second->isControllable(desired.parameter()))
+            mooseError("The desired parameter is not controllable: " << desired);
+
+          // Store pointer to the writable parameter
+          output.insert(desired, iter->second);
+
+          if (mark_as_controlled && tid == 0)
+            _controlled_parameters[iter->second].insert(desired);
```

Comment:
The `_controlled_parameters` storage is only for showing what parameters have been controlled for Console output purposes, so including parameters from each thread (which should always be named the same) would just make the output impossible to read.